stk/utilities/utilities.py

```diff
@@ -1143,10 +1146,438 @@ def vector_theta(vector1, vector2):
+        Extracts total energy in a.u. from xTB output.
+
+        Formatting based on latest version of xTB (190418)
+        Example line:
+        "          | TOTAL ENERGY              -76.260405590154 Eh   |"
+
+        Returns
+        -------
+        :class:`float`
+            Total energy in a.u.
+        """
+        value = None
+
+        # Regular expression for numbers.
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '          | TOTAL ENERGY  ' in line:
+                value = nums.search(line.rstrip()).group(0)
+                break
+
+        return float(value)
+
+    def homo_lumo_gap(self):
+        """
+        Extracts HOMO-LUMO gap in eV from xTB output.
+
+        Formatting based on latest version of xTB (190418)
+        Example line:
+        "          | HOMO-LUMO GAP               2.336339660160 eV   |"
+
+        Returns
+        -------
+        :class:`float`
+            Homo-Lumo gap in eV.
+        """
+        value = None
+
+        # Regular expression for numbers.
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '          | HOMO-LUMO GAP   ' in line:
+                value = nums.search(line.rstrip()).group(0)
+                break
+
+        return float(value)
+
+    def fermi_level(self):
+        """
+        Extracts Fermi-level energy in eV from xTB output.
+
+        Formatting based on latest version of xTB (190418)
+        Example line:
+        "             Fermi-level           -0.3159871 Eh
+                   -8.5984 eV"
+
+        Returns
+        -------
+        :class:`float`
+            Fermi-level in eV.
+        """
+        value = None
+
+        # Regular expression for numbers.
+        nums = re.compile(r"[+-]?\d+(?:\.\d+)?(?:[eE][+-]?\d+)?")
+        for line in reversed(self.output_string):
+            if '             Fermi-level        ' in line:
+                part2 = line.split('Eh')
+                value = nums.search(part2[1].rstrip()).group(0)
+                break
+
+        return float(value)
+
+    def qonly_dipole_moment(self):
+        """
+        Extracts `q only` dipole moment in Debye from xTB output.
+
+        Formatting based on latest version of xTB (190418)
+        Example line:
+        " q only:       -0.033      -0.081      -0.815"
+
+        Returns
+        -------
+        :class:`list` of :class:`float`
+            Components of dipole moment as ``[`x`, `y`, `z`]`` in a
+            list of length 3.
+
+        """
+        value = None
+
+        sample_set = []
+        for i, line in enumerate(self.output_string):
+            if 'molecular dipole:' in line:
+                sample_set = self.output_string[i+2].rstrip()
+
+        # get values from line
+        if 'q only:' in sample_set:
+            x, y, z = [i for i in sample_set.split(':')[1].split(' ')
```

Comment:
this list comprehension is incorrectly indented and can be written more cleanly by creating the float right away  

```python

return [
    float(i)
    for i in sample.set(':')[1].split(' ') if i
]

```
plus you can return right away as well

Reply:
Note that you dont need to compare to empty string because any 
if somestring

is true, for any string that is not the empty string and false for the empty string.